### PR TITLE
TASK: Contain dimension changes in tests to test case only

### PR DESCRIPTION
**Pull request by kitsunet**
The reset to empty array was technically wrong because dimensions were
configured. While this is not an issue at this time, it can be one
when other tests rely on the integrity of configured dimensions and
the repository.

### CODE REVIEW

**Comment by kitsunet** [comment]
For context, discussion from Slack

> The content repository tests reset the dimension configuration in the `\Neos\ContentRepository\Domain\Repository\ContentDimensionRepository`
itnerestingly we have multiple ways to find out about content dimensions (edited) 
the other ways still work
now the neos tests render a page
that all works fine
even the dimension rounting for the urls
but the alternate language links from the seo package which get added to page automatically try to fetch the node in another dimension
for the alternate links code verything seems fine, language dimension found
but the contextfactory then tries to create the context, checks with the dimension repository from above
and throws an exception because the dimension cannot be found (because it was removed in the repository tests)
tl;dr; repository tests do bad things, we just didn't have code that exposed the problem so far

This is all related to 4.3 and master branches but as the underlaying error (the spill from these tests) exists since ever we fix it here too.

**Comment by gerhard-boden** [comment]
now  that empty domain bug hit us again 🙈 
https://travis-ci.org/neos/neos-development-collection/jobs/516225978

**Comment by kitsunet** [comment]
I just included the other test change into here, and given that it's finally just test adjustments I am merging this.
